consumer.go

```diff
@@ -376,11 +393,13 @@ func (child *PartitionConsumer) Close() error {
 	// close itself)
 	close(child.dying)
```

Comment:
for more parallels with the producer, we should move this to an `AsyncClose` method